embedded_fab/main.dart

```diff
@@ -0,0 +1,416 @@
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:flutter/material.dart';
+
+import 'bottom_items.dart';
+import 'colors.dart';
+import 'content.dart' as Content;
+import 'notch.dart';
+import 'painters.dart';
+import 'shape_type.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      debugShowCheckedModeBanner: false,
+      home: Home(),
+      theme: ThemeData.light().copyWith(scaffoldBackgroundColor: Colors.white),
+    );
+  }
+}
+
+class Home extends StatefulWidget {
+  Home({Key key}) : super(key: key);
+
+  @override
+  _HomeState createState() => _HomeState();
+}
+
+enum Layout {
+  AppBar,
+  Experimental,
+  SideFAB,
+}
+
+class _HomeState extends State<Home> {
+  Color _selectedColor = Colors.white;
+  int _selectedIndex = 0;
+  Layout _layout = Layout.Experimental;
+  ShapeType _selectedShape = ShapeType.Cut;
+
+  Color get _strokeColor => _selectedColor == blue500 ? Colors.white : null;
+
+  FloatingActionButton _traditionalFAB(onTap) {
+    return FloatingActionButton(
+      backgroundColor: Colors.white,
+      child: Image.asset('assets/icon.png'),
+      onPressed: onTap,
+    );
+  }
+
+  void _onItemTapped(int index) {
+    setState(() {
+      _selectedIndex = index;
+    });
+  }
+
+  void _setSelectedTile(value) {
+    setState(() {
+      _selectedShape = value;
+    });
+  }
+
+  void _setSelectedColorTile(value) {
+    setState(() {
+      _selectedColor = value;
+    });
+  }
+
+  void _setLayout(value) {
+    setState(() {
+      _layout = value;
+    });
+  }
+
+  List<Widget> _itemOptions() {
+    return [
+      BottomNavItem(
+        icon: Icons.inbox,
+        isSelected: _selectedIndex == 0,
+        title: 'Inbox',
+        onTap: () {
+          _onItemTapped(0);
+        },
+      ),
+      BottomNavItem(
+        icon: Icons.star,
+        isSelected: _selectedIndex == 1,
+        title: 'Starred',
+        onTap: () {
+          _onItemTapped(1);
+        },
+      ),
+      SizedBox(width: 38),
+      BottomNavItem(
+        icon: Icons.send,
+        isSelected: _selectedIndex == 2,
+        title: 'Sent',
+        onTap: () {
+          _onItemTapped(2);
+        },
+      ),
+      BottomNavItem(
+        icon: Icons.archive,
+        isSelected: _selectedIndex == 3,
+        title: 'Archived',
+        onTap: () {
+          _onItemTapped(3);
+        },
+      )
+    ];
+  }
+
+  Widget _experimentalBottomNav() {
+    PlusPainter plusPainter;
+    switch (_selectedShape) {
+      case ShapeType.Bump:
+        {
```

Comment:
nit: the curly braces on each case statement are optional.

Reply:
Done